### PR TITLE
Fixed heartbeat

### PR DESCRIPTION
**Pull request by TarasLevelUp**
Fixed heartbeat logic to be more accurate on interval.
Fixed great number of tracebacks in tests for not closed heartbeat.

### CODE REVIEW

**Comment by TarasLevelUp** [comment]
Please don't set `_log_destroy_pending ` to False... It was not intended to be used that way. We need to fix problems, not hide them =(

**Comment by benjamin-hodgson** [comment]
There are a large number of unit tests which leave tasks incomplete because (for the purposes of the test) they're not interested in what happens when the task completes. They're unit tests, so it's fine. If you have a better suggestion for how to automatically close down all open tasks during a test then I'm listening :ear:

File: src/asynqp/connection.py [hunk 1]
```diff
@@ -64,6 +64,11 @@ def close(self):
         self.sender.send_Close(0, 'Connection closed by application', 0, 0)
         yield from self.synchroniser.await(spec.ConnectionCloseOK)
         self.closed.set_result(True)
+        # Close heartbeat
+        # TODO: We really need a better solution for finalization of parts
+        #       in the library.
+        self.protocol.heartbeat_monitor.stop()
+        yield from self.protocol.heartbeat_monitor.wait_closed()
```
Comment (benjamin-hodgson) [inline src/asynqp/connection.py]: I think this needs to be before `self.closed.set_resut(True)` above

File: test/heartbeat_tests.py [hunk 2]
```diff
@@ -17,13 +14,13 @@ def it_should_send_the_heartbeat(self):
 
     def cleanup_tasks(self):
         self.protocol.heartbeat_monitor.stop()
+        self.loop.run_until_complete(
+            self.protocol.heartbeat_monitor.wait_closed())
```
Comment (benjamin-hodgson) [inline test/heartbeat_tests.py]: We should wrap this in a timeout. If `wait_closed` hangs then the test will hang.

**Comment by TarasLevelUp** [comment]
> If you have a better suggestion for how to automatically close down all open tasks at the end of a test  then I'm listening :ear:
Close the coroutines by hand even in tests and wait for them to be closed (this way we will see if any of them raised an error). You will need to save all async tasks in code anyway, cause you want to log if some of them fail.

**Pull request merged**